File: cpp/cpptest/insertsort.cpp
#include<iostream>
using namespace std;

void insertsort(int arr[],int n){
    for (int i = 1; i < n;i++){
        int j;
        int temp = arr[i];
        for (j = i - 1; arr[j] > temp;j--){
            arr[j + 1] = arr[j];
        }
        arr[j + 1] = temp;
    }
}

int main(){
    int arr[] = {13, 12, 11, 10, 8, 5, 2, 20};
    int n = sizeof(arr) / sizeof(arr[0]);
    insertsort(arr, n);
    for (int i = 0; i < n;i++){
        cout << arr[i] << " ";
    }
    return 0;
}

File: python/pythontest/InsertSort.py
def insertsort(list):
    for i in range(1,len(list)):
        temp = list[i]
        j = i-1
        while j>=0 and list[j]>temp:
            list[j+1] = list[j]
            j = j-1
        list[j+1] = temp
        print(arr)

arr = [5,4,3,2,1]
insertsort(arr)